### ISSUE TITLE
Use specific Errors instead of generic Errors in Go

### ISSUE THREAD

**Issue by cmgriffing** [Vexilla/vexilla#50]
We are currently using generic errors in the Go SDK. It would be nice to be able to use `errors.Is` to detect specific errors as a consumer of the SDK.

**Comment by DirkHeijnen** [comment]
I'm onto it!

**Comment by DirkHeijnen** [comment]
Opened a PR for this issue. https://github.com/Vexilla/vexilla/pull/52

According to the answer on this [stackoverflow post](https://stackoverflow.com/questions/23796543/go-checking-for-the-type-of-a-custom-error) it seems that  `errors.Is` only checks of value equality and `errors.As` checks for type equality.

However unfortunately I can't get the `errors.As` to work as I would like.
![image](https://github.com/Vexilla/vexilla/assets/36156996/e1070fd9-4948-43c0-8949-7fe8abbf29e8)


Regular type equality checking works fine with:
```golang
func TestGroupLookupError(t *testing.T) {
	groupNameOrId := "somegroup"

	err := NewGroupLookupError(groupNameOrId)

	if _, ok := err.(*GroupLookupError); !ok {
		t.Errorf("Expected error type to be GroupLookupError but got %T", err)
	}
}

func TestDifferentErrorTypesAreNotTheSameWhenTypeChecking(t *testing.T) {
	groupNameOrId := "somegroup"
	environmentNameOrId := "someenvironment"

	groupError := NewGroupLookupError(groupNameOrId)
	environmentError := NewEnvironmentLookupError(groupNameOrId, environmentNameOrId)

	if _, ok := groupError.(*EnvironmentLookupError); ok {
		t.Errorf("Expected error type to be different but got %T", groupError)
	}

	if _, ok := environmentError.(*GroupLookupError); ok {
		t.Errorf("Expected error type to be different but got %T", environmentError)
	}
}
```

**Comment by cmgriffing** [comment]
This issue closed by #52
